src/compiler/checker.ts

```diff
@@ -27266,11 +27295,36 @@ namespace ts {
             node: PropertyAccessExpression | QualifiedName | PropertyAccessExpression | VariableDeclaration | ParameterDeclaration | ImportTypeNode | PropertyAssignment | ShorthandPropertyAssignment | BindingElement,
             isSuper: boolean, writing: boolean, type: Type, prop: Symbol, reportError = true): boolean {
 
-            const flags = getDeclarationModifierFlagsFromSymbol(prop, writing);
             const errorNode = node.kind === SyntaxKind.QualifiedName ? node.right :
                 node.kind === SyntaxKind.ImportType ? node :
                 node.kind === SyntaxKind.BindingElement && node.propertyName ? node.propertyName : node.name;
 
+            const typeChecks = checkPropertyAccessibilityAtNode(node, isSuper, writing, type, prop);
+            switch (typeChecks.result) {
+                case (TypeChecks.Ok):
+                    return true;
+                case (TypeChecks.HasDiagnostic):
+                    if (reportError) {
+                        error(errorNode, typeChecks.message, ...typeChecks.args);
+                    }
+                    return false;
+            }
+        }
+
+        // Possible concerns:

```

Comment:
left a comment with questions here instead of the usual jsdoc

Reply:
Would it be correct to immediately return false if node isn't one of the currently-required types?

On the other hand, it's tempting to specify a Node subtype, but not necessary a lot of the time. This could be one of those times. If it's not going to crash or run slowly, I'd say it's an improvement to accept any `Node`.

Reply:
I don't think it would crash. The specific `Node` subtypes were being used, as far as I could tell, for getting the error node to report diagnostics. And from what I can immediately see, I don't think there would be an impact for performance either, but I'm less confident on that because performance is very hard to access just by looking at the code...